runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -616,25 +633,62 @@ private void syncStreamUnsafe(ISMRStream stream, long timestamp) {
         log.trace("Sync[{}] {}", this, (timestamp == Address.OPTIMISTIC)
                 ? "Optimistic" : "to " + timestamp);
         long syncTo = (timestamp == Address.OPTIMISTIC) ? Address.MAX : timestamp;
-        stream.streamUpTo(syncTo)
-                .forEachOrdered(entry -> {
+
+        applyUpdateStreamUnsafe(stream.streamUpTo(syncTo), timestamp == Address.OPTIMISTIC);
+    }
+
+    private void applyUpdateStreamUnsafe(Stream<SMRRecord> smrRecordStream, boolean fromWriteSMRStream) {
```

Comment:
Can you rename `fromWriteSMRStream` to `isOptimistic`?

Reply:
done.